tracer/test/Datadog.Trace.Tests/DataStreamsMonitoring/DDSketchPoolTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// <copyright file="DDSketchPoolTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.DataStreamsMonitoring.Utils;
+using Datadog.Trace.Vendors.Datadog.Sketches;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.Tests.DataStreamsMonitoring;
+
+public class DDSketchPoolTests
+{
+    [Fact]
+    public void CanGetAndReleaseSketchRepeatedly()
+    {
+        var pool = new DDSketchPool();
+        var sketch = pool.Get();
+        pool.Release(sketch);
+
+        var sketch2 = pool.Get();
+        sketch2.Should().Be(sketch);
+        pool.Release(sketch2);
+
+        var sketch3 = pool.Get();
+        sketch2.Should().Be(sketch);
+        pool.Release(sketch3);
+    }
+
+    [Fact]
+    public void WhenPoolFull_GetsNewInstance()
```

Comment:
I'm not sure the test is fully testing what it declares. I think to fully test it, you'd need to do something like the following:
1. for 0..maxPoolSize + 1, create a new DDSketch object, save the object reference into your `pools` list, and call `pool.Release(sketch)`. As a result, the DDSketchPool will have a pool of all but the `maxPoolSize + 1` sketch
2. for 0..maxPoolSize + 1, call `pool.Get()`
3. Assert that the first `maxPoolSize` values are the same and the last sketch values are not

Reply:
I don't want to add clutter, so I'll clarify here: I think when you create a new DDSketch object, you could continue to use `pool.Get()` with the later assertion of `pools.Should().OnlyHaveUniqueItems();` to demonstrate the empty scenario, while releasing everything to the pool will demonstrate the full scenario

Reply:
You're right, my original test wasn't right at all. I have rewritten to use your suggestion (though I didn't put any assertions around the order of the items from the pool, so we can switch to a stack/array etc if needs be without breaking these tests) in https://github.com/DataDog/dd-trace-dotnet/pull/3177/commits/bf7a4696e5ca83628824623ace1903fba32ecf1f